lib/active_model/serializer.rb

```diff
@@ -199,16 +199,22 @@ def serializer_from_options(options)
       opts
     end
 
+    def self.serializers_cache
+      @serializers_cache ||= ThreadSafe::Cache.new
```

Comment:
I don't think that it will cause any memory issues. It is only storing mappings between existing model classes and serializer classes, and keeping the those classes in memory is going to use more memory that the mappings. 

Rails already uses `ThreadSafe::Cache`s to store some similar things like constant lookups and controller names.